core/src/main/java/com/linecorp/armeria/internal/RequestContextUtil.java

```diff
@@ -87,5 +87,20 @@ public static SafeCloseable pushWithRootAndOldCtx(ClientRequestContext currentCt
         }
     }
 
+    /**
+     * An {@link IllegalStateException} which is raised when pushing a context from the unexpected thread
+     * or forgetting to close the previous context.
+     */
+    public static final class IllegalContextPushingException extends IllegalStateException {
+
+        private static final long serialVersionUID = 5431942355463120798L;
+
+        public IllegalContextPushingException(RequestContext newCtx, RequestContext oldCtx) {
```

Comment:
This type looks good but just to clarify, I was just expecting the old factory method to return `IllegalStateException` instead of throwing it ;)

Reply:
Ah, I misunderstood. 😅